ldm/invoke/generator/diffusers_pipeline.py

```diff
@@ -400,7 +400,7 @@ def device(self) -> torch.device:
     @property
     def _submodels(self) -> Sequence[torch.nn.Module]:
         module_names, _, _ = self.extract_init_dict(dict(self.config))
-        values = [getattr(self, name) for name in module_names.keys()]
+        values = [getattr(self, name, getattr(self.config, name)) for name in module_names.keys()]
```

Comment:
I think it's correctly to do:
`getattr(self, name) if hasattr(self, name) else getattr(self.config, name)`
But... maybe it's better to do not in short form then?
```python
submodels = []
for name in module_names.keys():
    if hasattr(self. name):
        value = getattr(self, name)
    else:
        value = getattr(self.config, name)
    if isinstance(value, torch.nn.Module):
        submodels.append(value)
return submodels
```
Maybe this method can be simplified, but I not sure - can modules be in config or only in self object or in both.

Reply:
Would this work with just dropping the attempt to getattr(self, name) and only use getattr(self.config, name)?

`values = [getattr(self.config, name)) for name in module_names.keys()]`

Based on  the [comments on the diffusers PR 2849](https://github.com/huggingface/diffusers/pull/2849#discussion_r1150529864) you mentioned, it sounds like they are trying to eliminate the possible redundancy of having to look for these values in both self attr and self.config attr, and move to only keeping in self.config.
mentioned:

Reply:
Test this - if take values only from configs, there other errors appears.
So - we need get value from both self and self.config.